ebmdatalab/generate-ebmdatalab-stats.py

```diff
@@ -97,35 +186,104 @@ def main(args):
     phase_counts = phase_counts.drop(labels='Not Applicable')
     ax = phase_counts.plot(kind='barh', ax=axes[0, 1])
     ax.set_title('Clinical trials phase')
-    
+
     # Plot study type
     # Only include study types used in >= 5 trials
     study_type_counts = trials_df['study_type'].value_counts(ascending=True)
     study_type_counts = study_type_counts[study_type_counts >= 5]
     ax = study_type_counts.plot(kind='barh', ax=axes[1, 0])
     ax.set_title('Clinical trials study type')
-    
+
     # Plot common interventions
     # Only include trials with an intervention and interventions in >= 10 trials
     intervention_counts = trials_df['intervention'].value_counts(ascending=True)
     intervention_counts = intervention_counts.drop(labels='No Intervention')
     intervention_counts = intervention_counts[intervention_counts >= 10]
     ax = intervention_counts.plot(kind='barh', ax=axes[1, 1])
     ax.set_title('Clinical trials common interventions')
-    
+
     fig.savefig(args.output_figure + '.png', bbox_inches = "tight")
     fig.savefig(args.output_figure + '.svg', bbox_inches = "tight")
-    
+
     print(f'Wrote {args.output_figure}.png and {args.output_figure}.svg')
-    
+
+    # Clean and separate the names of each country in single-country and
+    # multi-country clinical trials. Single-country trials have only a single
+    # name (string) in the `countries` field. Multi-country trials have
+    # multiple comma-separated names.
+    # Drop 1 trial that lists every country.
+    valid_country = trials_df[trials_df['countries'] != "No Country Given"]
+    valid_country = valid_country[valid_country['trial_id'] != "ISRCTN80453162"]
+    single_countries = valid_country['countries'][~valid_country['countries'].str.contains(',')]
+    multi_countries = valid_country["countries"][valid_country["countries"].str.contains(',')]
+    multi_countries = pd.Series(
+        [country for country_list in multi_countries.str.split(',') for country in country_list]
+    )
+
+    # Identify the 3-letter ISO codes for each unique country
+    # Remove any leading/trailing whitespace that may result from splitting above
+    unique_countries = single_countries.append(multi_countries).str.strip().drop_duplicates()
+    country_codes = pd.DataFrame.from_dict(assign_ISO(unique_countries),
+                                           orient="index",
+                                           columns=["iso_a3"])
+
+    # Map the ISO codes onto the country data and count the frequency
+    single_countries_codes = pd.DataFrame(single_countries,
+                                          index=single_countries).join(country_codes)["iso_a3"]
+    single_countries_codes = single_countries_codes.dropna()
```

Comment:
You're using the 3 letter country code here, so it looks like we won't have a problem with Namibia (2 letter code `NA`), right?

h/t https://twitter.com/trang1618/status/1281339098956009472

Reply:
Oh wow!!!! I think it should be ok, it looks like alpha_3='NAM'. But that is a really good issue to be aware of 😅